File: Assets/L_Lua/Test/TestViewSystem.lua
local M = class("TestViewSystem",ViewSystem);


function M:RegistViewCompoent(viewCompoent)
	viewCompoent.CacheGameObjet = true;
	viewCompoent.CachePrefab = true;
	viewCompoent.layer = ViewLayer.TOP;
end


function M:GetModuleID()
	return MODULE_ID.TEST_MODULE
end

function  M:GetListener()
	return {MODULE_EVENT.SHOW_TEST_VIEW,MODULE_EVENT.CLOSE_TEST_VIEW}
end

function M:Notify(op,param )

	if op == MODULE_EVENT.SHOW_TEST_VIEW then
		self:ShowView(param)
	elseif op == MODULE_EVENT.CLOSE_TEST_VIEW then
		print("关闭View!!!");
	end

end


function M:ShowView(moduleEntity)

	local object = moduleEntity:Get(TestViewObject);
	local heroInfo = moduleEntity:Get(HeroInfo);
	local personalInfo = moduleEntity:Get(PersonalInfo);

	object.TextName.text = heroInfo.Name;
    object.TextAge.text = heroInfo.Age;
    object.TextGender.text = heroInfo.Gender;
    object.TextNation.text = heroInfo.Nation;

    object.TextEducation.text = personalInfo.Education;
    object.TextPower.text = personalInfo.Power;
    object.TextIntelligence.text = personalInfo.Intelligence;
    object.TextAgile.text = personalInfo.Agile;
end



return M;

File: Assets/Script/TestModule/TestViewSystem.cs
/****************************************************
 *  Copyright © 2018-2020 #AUTHORNAME#. All rights reserved.
 *------------------------------------------------------------------------
 *  文件：TestViewSystem.cs
 *  作者：Lonely
 *  日期：Created by #CreateTime#
 *  功能：Nothing
*****************************************************/

using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using Cs;

public class TestViewSystem : ViewSystem
{
    public override void RegistViewCompoent(ViewSystemCompoent viewCompoent)
    {
        viewCompoent.CacheGameObjet = false;
        viewCompoent.CachePrefab = false;
        viewCompoent.layer = ViewLayer.TOP;   
    }


    public override void Notify(int op, object param)
    {
        switch (op)
        {
            case (int)TEST_MODDULE_EVENT.SHOW_TEST_VIEW:
                {
                    ShowView((TestModuleEntity)param);
                }
                break;
        }
    }


    void ShowView(TestModuleEntity moduleEntity)
    {
        TestViewObject @object = moduleEntity.testViewObject;
        HeroInfo heroInfo = moduleEntity.heroInfo;
        PersonalInfo personalInfo = moduleEntity.personalInfo;

        @object.TextName.text = heroInfo.Name;
        @object.TextAge.text = heroInfo.Age.ToString();
        @object.TextGender.text = heroInfo.Gender.ToString();
        @object.TextNation.text = heroInfo.Nation;

        @object.TextEducation.text = personalInfo.Education;
        @object.TextPower.text = personalInfo.Power.ToString();
        @object.TextIntelligence.text = personalInfo.Intelligence.ToString();
        @object.TextAgile.text = personalInfo.Agile.ToString();
    }

    #region 事件属性

    public override int GetModuleID()
    {
        return (int)ModuleID.TestModule;
    }

    public override int[] GetListener()
    {
        return new int[] { (int)TEST_MODDULE_EVENT.SHOW_TEST_VIEW };
    }

    #endregion


}
